src/llama_stack_client/lib/agents/agent.py

```diff
@@ -19,12 +23,76 @@ def __init__(
         client: LlamaStackClient,
         agent_config: AgentConfig,
         custom_tools: Tuple[CustomTool] = (),
+        memory_bank_id: Optional[str] = None,
     ):
         self.client = client
         self.agent_config = agent_config
         self.agent_id = self._create_agent(agent_config)
         self.custom_tools = {t.get_name(): t for t in custom_tools}
         self.sessions = []
+        self.memory_bank_id = memory_bank_id
+
+    @classmethod
+    def with_memory(
+        cls,
+        client: LlamaStackClient,
+        agent_config: AgentConfig,
+        custom_tools: Tuple[CustomTool] = (),
+        memory_bank_id: Optional[str] = None,
+    ) -> "Agent":
+        if memory_bank_id is None:
+            memory_bank_id = "memory_bank_" + str(uuid.uuid4())
+            client.memory_banks.register(
+                memory_bank_id=memory_bank_id,
+                params={
+                    "memory_bank_type": "vector",
+                    "embedding_model": "all-MiniLM-L6-v2",
```

Comment:
do we want to hardcode the embedding model? Is this model available by default in all distributions?

Reply:
Yes, its available by default. but might be better to pick an embedding model that is availble. Let me change this.